src/dowel/tabular.py

```diff
@@ -0,0 +1,45 @@
+"""A `dowel.logger` input for tabular (key-value) data."""
+import numpy as np
+from tabulate import tabulate
+
+
+class Tabular:
```

Comment:
hmm so what does this do that a `dict` doesn't?

i see the `as_primitive_type` business, but didn't you already restrict the types that end up in this thing based on the `types_accepted` on your LogOutput? If an LogOutput can't output a FooType kv-pair, it will just not declare it in `types_accepted`.